postmark.php

```diff
@@ -196,6 +246,91 @@ function settings_html() {
     }
 }
 
+// Creates a table for storing logs of send attempts.
+function pm_log_create_db() {
+
+  global $wpdb;
+
+  $table_name = $wpdb->prefix . 'postmark_log';
+
+  // Creates a logs table if it doesn't exist already.
+  if( $wpdb->get_var("SHOW TABLES LIKE '$table_name'") != $table_name ) {
+
+    $charset_collate = $wpdb->get_charset_collate();
+
+    $sql = "CREATE TABLE $table_name (
+     id mediumint(9) NOT NULL AUTO_INCREMENT,
+     time datetime DEFAULT '0000-00-00 00:00:00' NOT NULL,
+     fromaddress text,
+     toaddress text,
+     subject text,
+     response text,
+     PRIMARY KEY  id (id)
+   ) $charset_collate;";
+
+    require_once( ABSPATH . 'wp-admin/includes/upgrade.php' );
+    dbDelta( $sql );
+
+    // Activates cron job for automatically purging old (7+ days) Postmark logs.
+    pm_log_cron_activation();
+
+  }
+}
+
+// Schedules cron job for deleting old Postmark logs.
+function pm_log_cron_activation() {
+  if ( !wp_next_scheduled( 'pm_log_cron_job' ) ) {
+    wp_schedule_event( time(), 'daily', 'pm_log_cron_job' );
+  }
+}
+
+// Attaches pm_clear_old_logs function to cron job hook.
+add_action('pm_log_cron_job', 'pm_clear_old_logs');
+
+// Deletes Postmark logs older than 7 days.
+function pm_clear_old_logs() {
+
+  global $wpdb;
+
+  $table_name = $wpdb->prefix . 'postmark_log';
+
+  // Deletes logs that are more than 7 days old.
+  $wpdb->query(
```

Comment:
You should Add a LIMIT to this clause and run it successively to avoid locking this table for too long, I'd go with something like 250 -> 500

Reply:
Added limit to avoid deleting more than 500 at a time.